pkg/agent/multicluster/mc_route_controller.go

```diff
@@ -213,20 +248,157 @@ func (c *MCDefaultRouteController) processNextWorkItem() bool {
 	}
 	defer c.queue.Done(obj)
 
-	if k, ok := obj.(string); !ok {
+	key, ok := obj.(string)
+	if !ok {
 		c.queue.Forget(obj)
 		klog.InfoS("Expected string in work queue but got", "object", obj)
 		return true
-	} else if err := c.syncMCFlows(); err == nil {
-		c.queue.Forget(k)
+	}
+
+	syncFn := func() error {
+		if c.wireGuardConfig != nil {
+			if err := c.reconcileWireGuard(); err != nil {
+				return err
+			}
+		}
+		return c.syncMCFlows()
+	}
+	if err := syncFn(); err == nil {
+		c.queue.Forget(key)
 	} else {
 		// Put the item back on the workqueue to handle any transient errors.
-		c.queue.AddRateLimited(k)
-		klog.ErrorS(err, "Error syncing key, requeuing", "key", k)
+		c.queue.AddRateLimited(key)
+		klog.ErrorS(err, "Error syncing key, requeuing", "key", key)
 	}
 	return true
 }
 
+// reconcileWireGuard reconciles WireGuard configurations in following way:
+//  1. If the current Node is the Multi-cluster Gateway Node, controller will try to initialize corresponding WireGuard
+//     configuration and route on the host, then add all existing ClusterInfoImports as WireGuard peers.
+//  2. If the current Node is not Multi-cluster Gateway Node, controller will try to clean up WireGuard.
+//
+// Note: MCDefaultRouteController runs only one worker to process Gateway and ClusterInfoImport. So we do not need
+// any synchronization mechanism.
+func (c *MCDefaultRouteController) reconcileWireGuard() error {
+	gateway, err := c.getActiveGateway()
+	if err != nil {
+		return err
+	}
+
+	amIGateway := gateway != nil && gateway.Name == c.nodeConfig.Name
+	if !amIGateway {
+		if c.wireGuardInitialized {
+			if err := c.cleanUpWireGuard(); err != nil {
+				return err
+			}
+			c.wireGuardInitialized = false
+		}
+		return nil
+	}
+	if !c.wireGuardInitialized {
+		if err := c.initializeWireGuard(); err != nil {
+			return err
+		}
+		c.wireGuardInitialized = true
+	}
+	ciImports, _ := c.ciImportLister.List(labels.Everything())
+	for _, ciImport := range ciImports {
+		if err := c.updateSingleWireGuardPeer(ciImport); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// updateSingleWireGuardPeer tries to update a WireGuard peer with ClusterInfoImport. If updating successfully,
+// it will also create host route the peer.
+func (c *MCDefaultRouteController) updateSingleWireGuardPeer(ciImport *mcv1alpha1.ClusterInfoImport) error {
+	klog.InfoS("Updating WireGuard peer with ClusterInfoImport", "ClusterInfoImport", ciImport)
```

Comment:
Better to use `klog.V(2).InfoS`, and print `klog.Kobj(ciImport)` only.

Reply:
Added log level.